kubeadm/flannel/setup.go

```diff
@@ -9,15 +9,19 @@ import (
 )
 
 func setupOverlay(interfaceName string) {
-	run(fmt.Sprintf(`ipmo C:\k\flannel\hns.psm1; New-HNSNetwork -Type Overlay -AddressPrefix "192.168.255.0/30"`+
-		`-Gateway "192.168.255.1" -Name "External" -AdapterName "%s" -SubnetPolicies @(@{Type = "VSID"; VSID = 9999; })`,
+	run(fmt.Sprintf(`ipmo C:\k\flannel\hns.psm1; `+
+                `if ((Get-HNSNetwork | ?{ ($_.Type -eq "Overlay") -and ($_.Name -eq "External") }) -eq $null) { `+
```

Comment:
If there is a network that is not overlay  with same name you will end up with the same error since it will try to create the network aftwards.  Someone might also have a bad day if that network named External is not of type "Overlay".  

Could you warn if the network type is not overlay? It might make sense to pull this out to a constant or use go templates to fill the values?

Reply:
Thank for review.
I added check for exist network type and warning about wrong type.
Please review again.

Reply:
> It might make sense to pull this out to a constant or use go templates to fill the values?
Maybe you are right but I have no go skills and tried to make minimal changes.